python/strings_udf/cpp/src/strings/udf/shim.cu

```diff
@@ -329,3 +330,20 @@ extern "C" __device__ int concat(int* nb_retval, void* udf_str, void* const* lhs
   *udf_str_ptr = result;
   return 0;
 }
+
+extern "C" __device__ int replace(
+  int* nb_retval, void* udf_str, void* const src, void* const to_replace, void* const replacement)
+{
+  auto src_ptr         = reinterpret_cast<cudf::string_view const*>(src);
+  auto to_replace_ptr  = reinterpret_cast<cudf::string_view const*>(to_replace);
+  auto replacement_ptr = reinterpret_cast<cudf::string_view const*>(replacement);
+
+  auto udf_str_ptr = new (udf_str) udf_string;
+  udf_string result;
+
+  result = replace(*src_ptr, *to_replace_ptr, *replacement_ptr);
+
+  *udf_str_ptr = result;
```

Comment:
The compiler may optimize this for us but changing to this _may_ prevent an extra alloc/copy:
```suggestion
  *udf_str_ptr = replace(*src_ptr, *to_replace_ptr, *replacement_ptr);
```